### PR TITLE
Added `setLogLevel`

### PR DESCRIPTION
**Pull request by NachoSoto**
- [x] Fixes [CSDK-609]
- [x] Blocked by [CSDK-549]
- [x] Depends on https://github.com/RevenueCat/purchases-hybrid-common/pull/297
- [x] Depends on https://github.com/RevenueCat/purchases-android/pull/753
- [x] Depends on https://github.com/RevenueCat/purchases-hybrid-common/pull/301

[CSDK-609]: https://revenuecats.atlassian.net/browse/CSDK-609?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CSDK-549]: https://revenuecats.atlassian.net/browse/CSDK-549?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/purchases.ts [hunk 1]
```diff
@@ -495,11 +510,20 @@ export default class Purchases {
...
      * Enables/Disables debugs logs
      * @param {boolean} enabled Enable or not debug logs
+     * @deprecated, use setLogLevel instead
      */
     public static async setDebugLogsEnabled(enabled: boolean): Promise<void> {
         RNPurchases.setDebugLogsEnabled(enabled);
     }
 
+    /**
+     * Used to set the log level. Useful for debugging issues with the lovely team @RevenueCat.
```
Comment (tonidero) [inline src/purchases.ts:520]: Might be good to mention what's the default (I believe it's `INFO`)

File: apitesters/purchases.ts [hunk 2]
```diff
@@ -124,6 +125,14 @@ async function checkConfigure() {
   const configured: boolean = await Purchases.isConfigured();
 }
 
+async function checkLogLevel() {
+  await Purchases.setLogLevel(LOG_LEVEL.DEBUG);
+  await Purchases.setLogLevel(LOG_LEVEL.VERBOSE);
+  await Purchases.setLogLevel(LOG_LEVEL.INFO);
+  await Purchases.setLogLevel(LOG_LEVEL.WARN);
+  await Purchases.setLogLevel(LOG_LEVEL.ERROR);
```
Comment (tonidero) [inline apitesters/purchases.ts:133]: Should we check the enum instead and just test this function with a parameter?

**Comment by vegaro** [comment]
I will add it to Android this week

File: src/purchases.ts (hunk 1 above)
Comment (NachoSoto) [reply src/purchases.ts:520]: 👍🏻

File: apitesters/purchases.ts (hunk 2 above)
Comment (NachoSoto) [reply apitesters/purchases.ts:133]: Good point.

**Comment by vegaro** [comment]
@NachoSoto I assigned you the Android task, since you're working on these it might be better for you to tackle that one

**Comment by NachoSoto** [comment]
This is ready!

File: __tests__/index.test.js [hunk 3]
```diff
@@ -315,7 +315,21 @@ describe("Purchases", () => {
...
+    Purchases.setLogLevel(Purchases.LOG_LEVEL.ERROR);
+
+    expect(NativeModules.RNPurchases.setLogLevel).toBeCalledWith("ERROR");
+    expect(NativeModules.RNPurchases.setLogLevel).toBeCalledTimes(1);
+
+    Purchases.setLogLevel(Purchases.LOG_LEVEL.VERBOSE);
+
+    expect(NativeModules.RNPurchases.setLogLevel).toBeCalledWith("VERBOSE");
+    expect(NativeModules.RNPurchases.setLogLevel).toBeCalledTimes(2);
+  });
```
Comment (vegaro) [inline __tests__/index.test.js:332]: Probably too much, but I'd like to mention in case it wasn't intentional. This is only testing for Error and Verbose.

**Review by vegaro — APPROVED** [review]

**Comment by vegaro** [comment]
Going to merge main into this branch because my branch depends on this change

File: __tests__/index.test.js (hunk 3 above)
Comment (NachoSoto) [reply __tests__/index.test.js:372]: Good point. I added tests for all levels.

**Pull request merged**